### ISSUE TITLE
loaFreeHistorySize isn't guarded completely. 

### ISSUE THREAD

**Issue by mgaudet** [eclipse/omr#1313]
`loaFreeHistorySize` is defined inside of  `GCExtensionBase.hpp` under `OMR_GC_MODRON_SCAVENGER`

However, it's used inside of `MemoryPoolLargeObjects.cpp` outside that guard, so if you build without `OMR_GC_MODRON_SCAVENGER` you'll see a failure like this: 

```
[ 12%] Building CXX object gc/CMakeFiles/omrgc.dir/base/MemoryPoolLargeObjects.cpp.o
/home/magaudet/open/omr/localizeCMake/gc/base/MemoryPoolLargeObjects.cpp: In member function ‘virtual bool MM_MemoryPoolLargeObjects::initialize(MM_EnvironmentBase*)’:
/home/magaudet/open/omr/localizeCMake/gc/base/MemoryPoolLargeObjects.cpp:116:73: error: ‘class MM_GCExtensionsBase’ has no member named ‘loaFreeHistorySize’
  _loaFreeRatioHistory = (double*)env->getForge()->allocate(_extensions->loaFreeHistorySize * sizeof(double), MM_AllocationCategory::FIXED, OMR_GET_CALLSITE());
                                                                         ^
/home/magaudet/open/omr/localizeCMake/gc/base/MemoryPoolLargeObjects.cpp:127:35: error: ‘class MM_GCExtensionsBase’ has no member named ‘loaFreeHistorySize’
  for (int i = 0; i < _extensions->loaFreeHistorySize; i++ ){
                                   ^
/home/magaudet/open/omr/localizeCMake/gc/base/MemoryPoolLargeObjects.cpp: In member function ‘double MM_MemoryPoolLargeObjects::calculateTargetLOARatio(MM_EnvironmentBase*, uintptr_t)’:
/home/magaudet/open/omr/localizeCMake/gc/base/MemoryPoolLargeObjects.cpp:371:28: error: ‘class MM_GCExtensionsBase’ has no member named ‘loaFreeHistorySize’
  for (int i = _extensions->loaFreeHistorySize; i > 0 ; i--){
                            ^
/home/magaudet/open/omr/localizeCMake/gc/base/MemoryPoolLargeObjects.cpp:380:97: error: ‘class MM_GCExtensionsBase’ has no member named ‘loaFreeHistorySize’
  _minLOAFreeRatio = *std::min_element(_loaFreeRatioHistory, _loaFreeRatioHistory + _extensions->loaFreeHistorySize);
                                                                                                 ^
gc/CMakeFiles/omrgc.dir/build.make:1094: recipe for target 'gc/CMakeFiles/omrgc.dir/base/MemoryPoolLargeObjects.cpp.o' failed
make[2]: *** [gc/CMakeFiles/omrgc.dir/base/MemoryPoolLargeObjects.cpp.o] Error 1
```

Pretty easy to reproduce from a fresh clone like this: 

`mkdir cmake_build && cd cmake_build && cmake .. && make`

**Comment by charliegracie** [comment]
@LinaSerry Can you take a look at this when you get a chance?  Thanks

**Comment by charliegracie** [comment]
Closing as I believe this was resolved by #1314
